src/marks/raster.js

```diff
@@ -0,0 +1,427 @@
+          .attr("xlink:href", canvas.toDataURL())
+      )
+      .node();
+  }
+}
+
+export function raster(data, options) {
+  if (arguments.length < 2) (options = data), (data = null);
+  let {x, y, fill, ...rest} = options;
+  // Because we implicit x and y when fill is a function of (x, y), and when
+  // data is a dense grid, we must further disambiguate by testing whether data
+  // contains [x, y, z?] tuples. Hence you can’t use this shorthand with a
+  // transform that lazily generates tuples, but that seems reasonable since
+  // this is just for convenience anyway.
+  if (x === undefined && y === undefined && isTuples(data)) {
+    (x = first), (y = second);
+    if (fill === undefined) fill = third;
+  }
+  return new Raster(data, {...rest, x, y, fill});
+}
+
+// Evaluates a function at pixel midpoints. TODO Faceting? Optimize linear?
+function sampleFill({fill, fillOpacity, pixelSize = 1, ...options} = {}) {
+  if (typeof fill !== "function") (options.fill = fill), (fill = null);
+  if (typeof fillOpacity !== "function") (options.fillOpacity = fillOpacity), (fillOpacity = null);
+  return initializer(options, (data, facets, {x1, y1, x2, y2}, {x, y}) => {
+    // TODO Allow projections, if invertible.
+    if (!x) throw new Error("missing scale: x");
+    if (!y) throw new Error("missing scale: y");
+    let {width: w, height: h} = options;
+    (x1 = x(x1.value[0])), (y1 = y(y1.value[0])), (x2 = x(x2.value[0])), (y2 = y(y2.value[0]));
+    // Note: this must exactly match the defaults in render above!
+    if (w === undefined) w = Math.round(Math.abs(x2 - x1) / pixelSize);
+    if (h === undefined) h = Math.round(Math.abs(y2 - y1) / pixelSize);
+    const kx = (x2 - x1) / w;
+    const ky = (y2 - y1) / h;
+    (x1 += kx / 2), (y1 += ky / 2);
+    let F, FO;
+    if (fill) {
+      F = new Array(w * h);
+      for (let yi = 0, i = 0; yi < h; ++yi) {
+        for (let xi = 0; xi < w; ++xi, ++i) {
+          F[i] = fill(x.invert(x1 + xi * kx), y.invert(y1 + yi * ky));
+        }
+      }
+    }
+    if (fillOpacity) {
+      FO = new Array(w * h);
+      for (let yi = 0, i = 0; yi < h; ++yi) {
+        for (let xi = 0; xi < w; ++xi, ++i) {
+          FO[i] = fillOpacity(x.invert(x1 + xi * kx), y.invert(y1 + yi * ky));
+        }
+      }
+    }
+    return {
+      data: F ?? FO,
+      facets,
+      channels: {
+        ...(F && {fill: {value: F, scale: "color"}}),
+        ...(FO && {fillOpacity: {value: FO, scale: "opacity"}})
+      }
+    };
+  });
+}
+
+function maybeRasterize(rasterize) {
+  if (typeof rasterize === "function") return rasterize;
+  if (rasterize == null) return rasterizeNone;
+  switch (`${rasterize}`.toLowerCase()) {
+    case "none":
+      return rasterizeNone;
+    case "dense":
+      return rasterizeDense;
+    case "nearest":
+      return rasterizeNearest;
+    case "barycentric":
+      return rasterizeBarycentric;
+    case "walk-on-spheres":
+      return rasterizeWalkOnSpheres;
```

Comment:
We should probably call this "random-walk" rather than "walk-on-spheres". It is more commonly known, and expresses well enough what's happening (as you put it: “if you were to be airdropped somewhere (…) and proceeded to walk randomly, this map tells you the expected [value] you’d run in to”). WoS converges much faster than a basic random walk would, and makes it feasible, but this is a question of implementation.